Sources/Sentry/SentryCrashReportConverter.m

```diff
@@ -316,21 +316,49 @@ - (SentryThread *_Nullable)crashedThread
     return [self threadAtIndex:self.crashedThreadIndex];
 }
 
-- (NSArray<SentryDebugMeta *> *)convertDebugMeta
+- (SentryDebugMeta *)debugMetaFromBinaryImageDictionary:(NSDictionary *)sourceImage
+{
+    SentryDebugMeta *debugMeta = [[SentryDebugMeta alloc] init];
+    debugMeta.uuid = sourceImage[@"uuid"];
+    debugMeta.type = @"apple";
+    // We default to 0 on the server if not sent
+    if ([sourceImage[@"image_vmaddr"] integerValue] > 0) {
+        debugMeta.imageVmAddress = sentry_formatHexAddress(sourceImage[@"image_vmaddr"]);
+    }
+    debugMeta.imageAddress = sentry_formatHexAddress(sourceImage[@"image_addr"]);
+    debugMeta.imageSize = sourceImage[@"image_size"];
+    debugMeta.name = sourceImage[@"name"];
+    return debugMeta;
+}
+
+- (NSArray<SentryDebugMeta *> *)debugMetaForThreads:(NSArray<SentryThread *> *)threads
 {
+    NSMutableSet<NSString *> *imageNames = [[NSMutableSet alloc] init];
+
+    for (SentryThread *thread in threads) {
+        for (SentryFrame *frame in thread.stacktrace.frames) {
+            if (frame.package && ![imageNames containsObject:frame.package]) {
+                [imageNames addObject:frame.package];
```

Comment:
`m`: Instead of `imageNames`, wouldn't it be better to store the `imageAddress` and then below lookup for `sentry_formatHexAddress(sourceImage[@"image_addr"])`?

Reply:
Thanks. 
Im my tests the package name is a file path, so it must be unique. But Im not sure if this is the case for every scenario. 
`imageAddress` is probably more safe.